Collection.php

```diff
@@ -844,21 +844,45 @@ public function buildCondition($condition)
             return [];
         }
         if (isset($condition[0])) { // operator format: operator, operand 1, operand 2, ...
-            $operator = strtoupper($condition[0]);
-            if (isset($builders[$operator])) {
+            $operator = $condition[0];
+            if (strncmp($operator, '$', 1) === 0) {
+                $method = 'buildSimpleCondition';
+            } elseif (isset($builders[strtoupper($operator)])) {
+                $operator = strtoupper($operator);
                 $method = $builders[$operator];
-                array_shift($condition);
-
-                return $this->$method($operator, $condition);
             } else {
                 throw new InvalidParamException('Found unknown operator in query: ' . $operator);
             }
+            array_shift($condition);
+            return $this->$method($operator, $condition);
         } else {
             // hash format: 'column1' => 'value1', 'column2' => 'value2', ...
             return $this->buildHashCondition($condition);
         }
     }
-
+    
+    /**
+     * Creates a condition on a column-value pair using an abritrary operator
+     * 
+     * @param string $operator The Mongo operator. Note that Mongo query operators must start with an $ (e.g. `$gt`, `$lt`, etc.)
+     * @param array $operands The column-value pair
+     * @return array The generated Mongo condition
+     * @throws InvalidParamException if $operator does not start with $ or when wrong number of operands have been given.
+     */
+    public function buildSimpleCondition($operator, $operands)
```

Comment:
Why not? The buildCondition method is a gateway method that calls specific conditionBuilders. Introducing the simple condition logic straight away into that method is not following single responsibility imho. But if it makes you happy I'll put the logic immediately in that method. Idk.  Let me know what you want and I'll change it.